skilloracle/endpoint/skilloracle.py

```diff
@@ -9,41 +9,54 @@
 except:
     from ..skilloracle import SkillOracle
 
-
-# This endpoint is WIP
 class SkillOracleEndpoint(object):
     def __init__(self, fetcher=None):
-        self.oracle = SkillOracle(port=7000)
+        self.host = "skilloracle"
+        self.oracle = SkillOracle(host=self.host, port=7000)
         self.put_valid_keys = {'label', 'name', 'context'}
         self.fetcher = fetcher
         if not fetcher:
             fetcher = None # what kind of default woudl we do here?
 
     def on_put(self, req, resp):
         query = falcon.uri.parse_query_string(req.query_string)
+        # ^ just use req.params.items or, below, req.params.keys()
         query_keys = set(query.keys())
 
         if self.put_valid_keys.issuperset(query_keys):
-            label = query['label']
-            name = query['name']
-            context = query['context']
+            print(req.params)
+            label = ''
+            name = ''
+            context = ''
+
+            if 'label' in req.params:
+                label = req.params['label'][0] # [0] if multiple values?
```

Comment:
This is not a blocker by any means, but one tweak could be making sure only one value is sent, so as to not confuse a user trying to use it in an unsupported way.